src/nbl/video/CVKLogicalDevice.h

```diff
@@ -75,44 +185,137 @@ class CVKLogicalDevice final : public ILogicalDevice
         return IGPUEvent::E_STATUS::ES_FAILURE;
     }
             
-    core::smart_refctd_ptr<IGPUFence> createFence(IGPUFence::E_CREATE_FLAGS _flags) override
+    core::smart_refctd_ptr<IGPUFence> createFence(IGPUFence::E_CREATE_FLAGS flags) override
     {
-        return nullptr;
+        VkFenceCreateInfo vk_createInfo = { VK_STRUCTURE_TYPE_FENCE_CREATE_INFO };
+        vk_createInfo.pNext = nullptr; // Each pNext member of any structure (including this one) in the pNext chain must be either NULL or a pointer to a valid instance of VkExportFenceCreateInfo or VkExportFenceWin32HandleInfoKHR
+        vk_createInfo.flags = static_cast<VkFenceCreateFlags>(flags);
+
+        VkFence vk_fence;
+        if (vkCreateFence(m_vkdev, &vk_createInfo, nullptr, &vk_fence) == VK_SUCCESS)
+        {
+            return core::make_smart_refctd_ptr<CVulkanFence>(
+                core::smart_refctd_ptr<CVKLogicalDevice>(this), flags, vk_fence);
+        }
+        else
+        {
+            return nullptr;
+        }
     }
             
     IGPUFence::E_STATUS getFenceStatus(IGPUFence* _fence) override
     {
         return IGPUFence::E_STATUS::ES_ERROR;
```

Comment:
cmon you can do it!

P.S. without this out streaming staging buffer will not work